worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,374 @@
+                "Macro Zone 3 - Fiery Mario-Special Agent Midway Bell", self.player),
+            "Macro Zone 3 - Fiery Mario-Special Agent Midway Bell": lambda state: (has_pipe_down(state, self.player)
+                and has_pipe_down(state, self.player)) or state.has(
+                "Macro Zone 3 - Fiery Mario-Special Agent Midway Bell", self.player),
+            "Macro Zone 4 - One Mighty Mouse": lambda state: has_pipe_right(state, self.player),
+            "Pumpkin Zone 1 - Bat Course": lambda state: has_pipe_down(state, self.player) or state.has(
+                "Pumpkin Zone 1 - Bat Course Midway Bell", self.player),
+            "Pumpkin Zone 1 - Bat Course Midway Bell": lambda state: has_pipe_down(state, self.player) or state.has(
+                "Pumpkin Zone 1 - Bat Course Midway Bell", self.player),
+            "Pumpkin Zone 2 - Cyclops Course": lambda state: has_pipe_down(state, self.player) and has_pipe_up(
+                state, self.player) and has_pipe_right(state, self.player) and state.has("Swim", self.player),
+            # You can only spin jump as Big Mario or Fire Mario
+            "Pumpkin Zone 2 - Cyclops Course Secret Exit": lambda state: has_pipe_down(
+                state, self.player) and has_pipe_up(state, self.player) and has_pipe_right(
+                state, self.player) and state.has("Swim", self.player) and state.has_any(
+                ["Mushroom", "Fire Flower"], self.player),
+            "Pumpkin Zone 3 - Ghost House Secret Exit": lambda state: state.has("Carrot", self.player),
+            "Pumpkin Zone 4 - Witch's Mansion": lambda state: has_pipe_right(state, self.player),
+            "Mario Zone 1 - Fiery Blocks": lambda state: has_pipe_right(state, self.player),
+            # It is possible to get as small mario, but it is a very precise jump and you will die afterward.
+            "Mario Zone 1 - Fiery Blocks Midway Bell": lambda state: (state.has_any(
+                ["Mushroom", "Fire Flower", "Carrot"], self.player) and has_pipe_right(state, self.player))
+                or state.has("Mario Zone 1 - Fiery Blocks Midway Bell", self.player),
+            "Mario Zone 4 - Three Mean Pigs!": lambda state: has_pipe_right(state, self.player),
+            "Turtle Zone 2 - Turtle Zone": lambda state: has_pipe_up(state, self.player) and has_pipe_down(
+                state, self.player) and has_pipe_right(state, self.player) and has_pipe_left(state, self.player)
+                and state.has("Swim", self.player),
+            "Turtle Zone 2 - Turtle Zone Midway Bell": lambda state: state.has_any(
+                ["Swim", "Turtle Zone 2 - Turtle Zone Midway Bell"], self.player),
+            "Turtle Zone 2 - Turtle Zone Secret Exit": lambda state: has_pipe_up(
+                state, self.player) and state.has("Swim", self.player), #state.has_any(["Swim", "Turtle Zone 2 - Turtle Zone Midway Bell"], self.player),  # hard logic option?
```

Comment:
Two spaces required before inline comment
```suggestion
                state, self.player) and state.has("Swim", self.player),  #state.has_any(["Swim", "Turtle Zone 2 - Turtle Zone Midway Bell"], self.player),  # hard logic option?
```